os/using-environment-variables-in-systemd-units.md

```diff
@@ -0,0 +1,116 @@
+# Using Environment Variables In systemd Units
+
+## Environment directive
+
+Systemd has Environment directive which sets environment variables for executed processes. It takes a space-separated list of variable assignments. This option may be specified more than once in which case all listed variables will be set. If the same variable is set twice, the later setting will override the earlier setting. If the empty string is assigned to this option, the list of environment variables is reset, all prior assignments have no effect.
+
+### etcd2.service Unit Example
+
+Let's review etcd2.service [drop-in] unit example. [Cloud-config][cloud-config] compiles special `/run/systemd/system/etcd2.service.d/20-cloudinit.conf` [drop-in] unit file using options defined in "etcd2:" yaml section: 
+
+```
+[Service]
+Environment="ETCD_ADVERTISE_CLIENT_URLS=http://10.0.1.10:2379"
+Environment="ETCD_DISCOVERY=https://discovery.etcd.io/<token>"
+Environment="ETCD_INITIAL_ADVERTISE_PEER_URLS=http://10.0.1.10:2380"
+Environment="ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379,http://0.0.0.0:4001"
+Environment="ETCD_LISTEN_PEER_URLS=http://0.0.0.0:2380"
+```
+
+For example you have created five-nodes CoreOS cluster but have forgot to set cluster size in [discovery][etcd-discovery] URL and its cluster size is three by default. Your rest two nodes became proxy and you would like to convert them to etcd2 member.
```

Comment:
Last thing: can we move that new example up to the top so you start with the simple one then go into the more complicated one?